src/test/enhetstester/kotlin/no/nav/familie/ks/sak/sikkerhet/TilgangServiceTest.kt

```diff
@@ -0,0 +1,502 @@
+import no.nav.familie.ks.sak.data.TestdataGenerator.lagPersonopplysningGrunnlag
+import no.nav.familie.ks.sak.integrasjon.familieintegrasjon.IntegrasjonService
+import no.nav.familie.ks.sak.kjerne.behandling.domene.BehandlingRepository
+import no.nav.familie.ks.sak.kjerne.fagsak.FagsakService
+import no.nav.familie.ks.sak.kjerne.personident.Aktør
+import no.nav.familie.ks.sak.kjerne.personident.Personident
+import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.domene.Kjønn
+import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.domene.Person
+import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.domene.PersonType
+import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.domene.PersonopplysningGrunnlag
+import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.domene.PersonopplysningGrunnlagRepository
+import no.nav.familie.log.mdc.MDCConstants
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
+import org.slf4j.MDC
+import org.springframework.cache.CacheManager
+import org.springframework.cache.concurrent.ConcurrentMapCacheManager
+import java.time.LocalDate
+
+@ExtendWith(MockKExtension::class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class TilgangServiceTest {
+
+    @MockK
+    private lateinit var integrasjonService: IntegrasjonService
+
+    @MockK
+    private lateinit var behandlingRepository: BehandlingRepository
+
+    @MockK
+    private lateinit var fagsakService: FagsakService
+
+    @MockK
+    private lateinit var personopplysningGrunnlagRepository: PersonopplysningGrunnlagRepository
+
+    private val cacheManager = ConcurrentMapCacheManager()
+
+    private val rolleConfig = RolleConfig(
+        BehandlerRolle.BESLUTTER.name,
+        BehandlerRolle.SAKSBEHANDLER.name,
+        BehandlerRolle.VEILEDER.name,
+        KODE6 = "kode6",
+        KODE7 = "kode7"
+    )
+    private val auditLogger = AuditLogger("familie-ks-sak")
+    private lateinit var tilgangService: TilgangService
+
+    private val fagsak = defaultFagsak()
+    private val behandling = lagBehandling(fagsak)
+    private val aktør = fagsak.aktør
+    private val personopplysningGrunnlag = lagPersonopplysningGrunnlag(
+        behandlingId = behandling.id,
+        søkerPersonIdent = aktør.aktivFødselsnummer(),
+        barnasIdenter = emptyList()
+    )
+
+    @BeforeAll
+    fun beforeAll() {
+        tilgangService = TilgangService(
+            integrasjonService = integrasjonService,
+            behandlingRepository = behandlingRepository,
+            personopplysningGrunnlagRepository = personopplysningGrunnlagRepository,
+            fagsakService = fagsakService,
+            rolleConfig = rolleConfig,
+            cacheManager = cacheManager,
+            auditLogger = auditLogger
```

Comment:
```suggestion
            integrasjonService = mockIntegrasjonService,
            behandlingRepository = mockBehandlingRepository,
            personopplysningGrunnlagRepository = mockPersonopplysningGrunnlagRepository,
            fagsakService = mockFagsakService,
            rolleConfig = rolleConfig,
            cacheManager = cacheManager,
            auditLogger = auditLogger
```

Hmm, siden TilgangService tar inn både ikke-mocks og mocks, så hadde det kanskje vært lurt å endre navn på mocksa til å være prefikset med mock. På den måten ser man klart hva som er hva 😛

Reply:
Veldig god ide! 👍 I fix